src/components/check-reauth-users/tests/check-reauth-user-service.test.ts

```diff
@@ -1,22 +1,66 @@
-import { expect } from "chai";
 import { describe } from "mocha";
-
-import { supportReauthentication } from "../../../config";
+import { Http } from "../../../utils/http";
+import { sinon } from "../../../../test/utils/test-utils";
+import { API_ENDPOINTS } from "../../../app.constants";
+import { SinonStub } from "sinon";
+import { checkReauthUsersService } from "../check-reauth-users-service";
+import { CheckReauthServiceInterface } from "../types";
+import {
+  checkApiCallMadeWithExpectedBodyAndHeaders,
+  commonVariables,
+  expectedHeadersFromCommonVarsWithoutSecurityHeaders,
+  resetApiKeyAndBaseUrlEnvVars,
+  setupApiKeyAndBaseUrlEnvVars,
+} from "../../../../test/helpers/service-test-helper";
 
 describe("re-authentication service", () => {
-  describe("with auth re-authentication feature flag on", () => {
-    it("should return true", async () => {
-      process.env.SUPPORT_REAUTHENTICATION = "1";
+  const httpInstance = new Http();
+  const service: CheckReauthServiceInterface =
+    checkReauthUsersService(httpInstance);
+  const SUBJECT = "123";
+  let postStub: SinonStub;
 
-      expect(supportReauthentication()).to.be.true;
-    });
+  beforeEach(() => {
+    setupApiKeyAndBaseUrlEnvVars();
+    postStub = sinon.stub(httpInstance.client, "post");
   });
 
-  describe("with auth re-authentication feature flag off", () => {
-    it("should return false", async () => {
-      process.env.SUPPORT_REAUTHENTICATION = "0";
+  afterEach(() => {
+    postStub.reset();
+    resetApiKeyAndBaseUrlEnvVars();
+  });
 
-      expect(supportReauthentication()).to.be.false;
+  it("successfully calls the API to check a reauth user", async () => {
+    const axiosResponse = Promise.resolve({
+      data: {},
+      status: 200,
+      statusText: "OK",
     });
+    postStub.resolves(axiosResponse);
+    const { sessionId, email, ip, clientSessionId, diPersistentSessionId } =
+      commonVariables;
+
+    const underTest = () =>
+      service.checkReauthUsers(
+        sessionId,
+        email,
+        SUBJECT,
+        ip,
+        clientSessionId,
+        diPersistentSessionId
+      );
+
+    const expectedApiCallDetails = {
+      expectedPath: API_ENDPOINTS.CHECK_REAUTH_USER,
+      expectedHeaders: expectedHeadersFromCommonVarsWithoutSecurityHeaders,
```

Comment:
Why do we need to pass this?  Maybe it could be a way to add additional headers to be checked rather simply passing a constant from the service-test-helper module back to it.